### PR TITLE
SLING-11374 - Improve the details of the exceptions thrown by the BinaryOperator

### PR DESCRIPTION
**Pull request by ashokmca07**

### CODE REVIEW

**Review by raducotescu — APPROVED** [review]
@ashokmca07, thanks for taking the time to look into this! Could you please update the parent POM in a different PR?

File: src/test/java/org/apache/sling/scripting/sightly/compiler/expression/nodes/BinaryOperatorTest.java [hunk 1]
```diff
@@ -106,4 +106,44 @@ public void testStrictEqWrongType() {
...
 
+    @RunWith(Parameterized.class)
+    public static class NumberError {
+
+        @Parameters(name = "Comparison: {2}")
+        public static Iterable<? extends Object> data() {
+            List<Object[]> list = new ArrayList<>();
+            list.add(new Object[]{1, new Object(), "number to object not equal"});
+            list.add(new Object[]{new Object(), 2, "object to number not equal"});
+            list.add(new Object[]{TestEnum.ONE, 1, "String to number not equal"});
```
Comment (raducotescu) [inline src/test/java/org/apache/sling/scripting/sightly/compiler/expression/nodes/BinaryOperatorTest.java:117]: Here I think the correct message is "enum to number not equal". You would need a different test for comparing a String to a number.

**Review by raducotescu — CHANGES_REQUESTED** [review]

File: src/test/java/org/apache/sling/scripting/sightly/compiler/expression/nodes/BinaryOperatorTest.java (hunk 1 above)
Comment (ashokmca07) [reply src/test/java/org/apache/sling/scripting/sightly/compiler/expression/nodes/BinaryOperatorTest.java]: @raducotescu: I have added test coverage for `string to number` and corrected the relevant messages

**Comment by ashokmca07** [comment]
Expected: FYI, test coverage will fail on PR's (https://issues.apache.org/jira/browse/SLING-11400)

**Review by raducotescu — APPROVED** [review]
LGTM, but please add the POM update in a dedicated PR (or directly on `master`, with its own issue - similar to https://issues.apache.org/jira/browse/SLING-11355).

**Pull request merged**